src/pages/index.js

```diff
@@ -2,107 +2,223 @@ import "../pages/index.css";
 import Card from "../components/Card.js";
 import FormValidator from "../components/FormValidator.js";
 import UserInfo from "../components/UserInfo.js";
-import PopupWithForm from "../components/PopupWithForm.js";
-import Popup from "../components/Popup.js";
 import Section from "../components/Section.js";
-import PopupWithImages from "../components/PopupWithImages.js";
-
+import PopupWithForm from "../components/PopupWithForm.js";
+import PopupWithImage from "../components/PopupWithImage.js";
+import PopupWithConfirm from "../components/PopupWithConfirm.js";
+import Api from "../components/Api.js";
 import {
-    settings,
-    initialCards,
-    profileEditButton,
-    profileEditModal,
-    profileModalCloseButton,
-    profileTitle,
-    profileDescription,
-    profileTitleInput,
-    profileDescriptionInput,
-    profileEditForm,
-    addCardButton,
-    addCardModal,
-    addImageCloseButton,
-    addCardTitle,
-    addCardURL,
-    addCardTitleInput,
-    addCardURLInput,
-    cardListEl,
-    addCardForm,
-    imageModal,
-    imageModalImg,
-    imageModalText,
-    imageModalCloseButton,
+  settings,
+  initialCards,
+  profileEditButton,
+  profileEditModal,
+  profileModalCloseButton,
+  profileTitle,
+  profileDescription,
+  profileTitleInput,
+  profileDescriptionInput,
+  profileEditForm,
+  addCardButton,
+  addCardModal,
+  addImageCloseButton,
+  addCardTitle,
+  addCardURL,
+  addCardTitleInput,
+  addCardURLInput,
+  cardListEl,
+  addCardForm,
+  imageModal,
+  imageModalImg,
+  imageModalText,
+  imageModalCloseButton,
+  avatarEditButton,
+  avatarEditForm,
 } from "../utils/constants.js";
 
-const userInfo = new UserInfo(".profile__title", ".profile__description");
-const imagePopup = new PopupWithImages({ modalSelector: "#preview-image-modal"});
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
+);
 
-profileEditButton.addEventListener("click", () => {
-    const userData = userInfo.getUserInfo();
-    profileTitleInput.value = userData.name;
-    profileDescriptionInput.value = userData.job;
-});
+const imagePopup = new PopupWithImage({ modalSelector: "#preview-image-modal" });
 
 function handleImageClick(name, link) {
-    const cardData = { name, link };
-    imagePopup.open(cardData);
+  const cardData = { name, link };
+  imagePopup.open(cardData);
 }
 
+function handleDeleteCard(card) {
+  deleteCardPopup.setSubmitFunc(() => {
+   api.deleteCard(card.getId())
```

Comment:
* I can't delete cards. See the errors https://snipboard.io/RKHy9B.jpg

`getId` doesn't exist inside the `Card` class. Please, create such a method there